application/auth/models.py

```diff
@@ -0,0 +1,66 @@
+import datetime
+import jwt
+
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask_login import UserMixin
+from flask import current_app
+
+from application import db
+
+class User(UserMixin, db.Model):
+    __tablename__ = "users"
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(64), unique=True, index=True)
+    password_hash = db.Column(db.String(128))
+    date_created = db.Column(db.DateTime, default=datetime.datetime.utcnow())
+
+    def __init__(self, username, password):
+        self.username = username
+        self.password = password
+
+    @property
+    def password(self):
+        raise AttributeError('password is not a readable attribute')
+
+    @password.setter
+    def password(self, password):
+        self.password_hash = generate_password_hash(password)
+
+    def verify_password(self, password):
+        return check_password_hash(self.password_hash, password)
+
+    def generate_auth_token(self, user_id):
+        """ Generates the access token"""
+
+        try:
+            # set up a payload with an expiration time
+            payload = {
+                'exp': datetime.datetime.utcnow() + datetime.timedelta(minutes=5),
```

Comment:
consider increasing the `exp` to about 30 mins to avoid constant login especially during the defence